frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java

```diff
@@ -46,9 +49,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       placeService.validatePlaceId(place.placesApiPlaceId());
 
       String tripId = ServletUtil.matchUriOrThrowError(request, TRIP_NAME_PATTERN).group(1);
-
-      PlaceVisitModel newPlace =
-          place.toBuilder().setId(UUID.randomUUID().toString()).setTripId(tripId).build();
+      Key parentKey = KeyFactory.createKey(TRIP_ME_OUT_KIND_NAME, tripId);
+      String id = KeyFactory.keyToString(
+          KeyFactory.createKey(parentKey, tripId, UUID.randomUUID().toString()));
+      PlaceVisitModel newPlace = place.toBuilder().setId(id).setTripId(tripId).build();
```

Comment:
The servlet shouldn't know about KeyFactory and stuff. Just use a UUID for new places. The old code was correct.

Reply:
See the issue is I need to do this in order to retrieve the placeVisit in the datastore. I couldn't figure out a different way to do this.

Reply:
If you have to do this to get the datastore-based storage implementation to work, there is something wrong with the datastore-based storage.